File: ios/ExpoMsalModule.swift
import ExpoModulesCore

public class ExpoMsalModule: Module {
  public func definition() -> ModuleDefinition {
    name("ExpoMsal")

    function("helloAsync") { (options: [String: String]) in
      print("Hello 👋")
    }

    viewManager {
      view {
        ExpoMsalView()
      }

      prop("name") { (view: ExpoMsalView, prop: String) in
        print(prop)
      }
    }
  }
}

File: android/src/main/java/energy/stash.msal/ExpoMsalModule.kt
package energy.stash.msal

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoMsalModule : Module() {
  override fun definition() = ModuleDefinition {
    name("ExpoMsal")

    function("helloAsync") { options: Map<String, String> ->
      println("Hello 👋")
    }

    viewManager {
      view { context -> 
        ExpoMsalView(context) 
      }

      prop("name") { view: ExpoMsalView, prop: String ->
        println(prop)
      }
    }
  }
}
